proteus/Transport.py

```diff
@@ -1493,21 +1493,14 @@ def findReferenceComponent_phi(diffusion):
         log(memory("TimeIntegration","OneLevelTransport"),level=4)
         log("Calculating numerical quadrature formulas",2)
         self.calculateQuadrature()
-        #lay out components/equations contiguously for now
-        self.offset = [0]
-        for ci in range(1,self.nc):
-            self.offset += [self.offset[ci-1]+self.nFreeDOF_global[ci-1]]
-        self.stride = [1 for ci in range(self.nc)]
-        #use contiguous layout of components for parallel, requires weak DBC's
+
         comm = Comm.get()
         self.comm=comm
         if comm.size() > 1:
             assert numericalFluxType != None and numericalFluxType.useWeakDirichletConditions,"You must use a numerical flux to apply weak boundary conditions for parallel runs"
-            self.offset = [0]
-            for ci in range(1,self.nc):
-                self.offset += [ci]
-            self.stride = [self.nc for ci in range(self.nc)]
-        #
+
+        self.setupInterleavedFieldStrides()
+
```

Comment:
I don't mind pushing this into a function, but we do  need to preserve the ability to do both layouts, particularly in Transport.py.  The contiguous approach you deleted will always work, while the interleaved approach will break for  strong Dirichlet conditions and mixed trial spaces for multi-component PDEs.